### ISSUE TITLE
UpdateSystemPackages should not return PackageMissingResult if latest package already there

### ISSUE THREAD

**Issue by abedford** [OpenWrap/openwrap#205]
When calling UpdateSystemPackage from DefaultPackageManager if the latest package is already in the system repository then the dependencies will not resolve to anything of higher version and so a PackageMissingResult will be returned. It would be more useful if you could add an AlreadyAtLatestVersionResult or similar so we can distinguish between the various scenarios.

**Comment by serialseb** [comment]
Hmm, it should deinitly not return a PackageMissingResult, it should do nothing at all. Normally PackageMissingResult does mean just that, when you call the code do you pass the system repository as one of the repositories to look for stuff in?

If you had a code repro that'd be great so we can fix that. Would also be useful to know which revision of 1.1 you guys have branched from, to see if the problem is not already solved  by some tweaks.

:)

**Comment by abedford** [comment]
I've got a test which reproduces the problem, see below. This will return PackageMissingResult on the UpdateSystemPackages call.  Looks like we're using openwrap-1.1.0.64779355.

```
var system = new InMemoryRepository("System");
var remote = new InMemoryRepository("Remote");
var packageInfo = new InMemoryPackage
        {
            Name = "App1",
            Version =Version.Parse("1.1.0.0"),
            Source = remote
        };
remote.Packages.Add(packageInfo);

var packageResolver = new ExhaustiveResolver();
var packageManager = new DefaultPackageManager(new DefaultPackageDeployer(),
                                                    packageResolver,
                                                    new DefaultPackageExporter(new IExportProvider[]
                                                              {
                                                                   new DefaultAssemblyExporter()
                                                              })
                                                    );

var request = PackageRequest.Exact("App1", Version.Parse("1.1.0.0"));
packageManager.AddSystemPackage(request, new[] { remote}, system);
var result = packageManager.UpdateSystemPackages(new[] { remote }, system,
                                                           request.Name); 
```

**Comment by serialseb** [comment]
Fixed in pull request, in 2.0 codebase.
